File: reference-code/src/main/java/createcommands/arguments/types/misc/AdvancementArguments.java
package createcommands.arguments.types.misc;

import dev.jorel.commandapi.CommandAPICommand;
import dev.jorel.commandapi.arguments.AdvancementArgument;
import dev.jorel.commandapi.arguments.PlayerArgument;
import org.bukkit.advancement.Advancement;
import org.bukkit.advancement.AdvancementProgress;
import org.bukkit.entity.Player;

class AdvancementArguments {
    static {
        // #region advancementArgumentsExample
        new CommandAPICommand("award")
            .withArguments(new PlayerArgument("player"))
            .withArguments(new AdvancementArgument("advancement"))
            .executes((sender, args) -> {
                Player target = (Player) args.get("player");
                Advancement advancement = (Advancement) args.get("advancement");

                // Award all criteria for the advancement
                AdvancementProgress progress = target.getAdvancementProgress(advancement);
                for (String criteria : advancement.getCriteria()) {
                    progress.awardCriteria(criteria);
                }
            })
            .register();
        // #endregion advancementArgumentsExample
    }
}

File: reference-code/src/main/kotlin/createcommands/arguments/types/misc/AdvancementArguments.kt
package createcommands.arguments.types.misc

import dev.jorel.commandapi.CommandAPICommand
import dev.jorel.commandapi.arguments.AdvancementArgument
import dev.jorel.commandapi.arguments.PlayerArgument
import dev.jorel.commandapi.executors.CommandExecutor
import dev.jorel.commandapi.kotlindsl.advancementArgument
import dev.jorel.commandapi.kotlindsl.anyExecutor
import dev.jorel.commandapi.kotlindsl.commandAPICommand
import dev.jorel.commandapi.kotlindsl.playerArgument
import org.bukkit.advancement.Advancement
import org.bukkit.entity.Player

fun advancementArguments() {
    // #region advancementArgumentsExample
    CommandAPICommand("award")
        .withArguments(PlayerArgument("player"))
        .withArguments(AdvancementArgument("advancement"))
        .executes(CommandExecutor { _, args ->
            val target = args["player"] as Player
            val advancement = args["advancement"] as Advancement

            // Award all criteria for the advancement
            val progress = target.getAdvancementProgress(advancement)
            for (criteria in advancement.criteria) {
                progress.awardCriteria(criteria)
            }
        })
        .register()
    // #endregion advancementArgumentsExample
}

fun advancementArgumentsDSL() {
    // #region advancementArgumentsExampleDSL
    commandAPICommand("award") {
        playerArgument("player")
        advancementArgument("advancement")
        anyExecutor { _, args ->
            val target = args["player"] as Player
            val advancement = args["advancement"] as Advancement

            // Award all criteria for the advancement
            val progress = target.getAdvancementProgress(advancement)
            for (criteria in advancement.criteria) {
                progress.awardCriteria(criteria)
            }
        }
    }
    // #endregion advancementArgumentsExampleDSL
}